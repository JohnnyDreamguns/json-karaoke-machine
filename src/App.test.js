import React from 'react';
import { 
  App, 
  mapStateToProps, 
  mapDispatchToProps,
} from './App';
import renderer from 'react-test-renderer';

const mockState = {
  settings: {
    tempo: 120,
    beatNumber: 0
  }
};

const mockDispatchMethods = {
  handleChangeTempo: () => {},
  setBeatNumber: () => {}
};

const mockEvent = {
  target:{
    value: 100
  }
};

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<App></App>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('to return the correct state', () => {
      expect(mapStateToProps(mockState)).toEqual(mockState.settings);
    });
  });

  describe('mapDispatchToProps', () => {
    it('to return the correct result', () => {
      expect(JSON.stringify(mapDispatchToProps()))
        .toEqual(JSON.stringify(mockDispatchMethods));
    });
  });

  describe('handleChangeTempo', () => {
    const mockDispatch = jest.fn(() => {});
    const { handleChangeTempo } = mapDispatchToProps(mockDispatch);
    
    it('calls dispatch method once', () => {
      handleChangeTempo(mockEvent);
      expect(mockDispatch.mock.calls.length).toBe(1);
    });

    it('dispatches action with correct value', () => {
      mockDispatch.mockReset();
      handleChangeTempo(mockEvent);
      expect(mockDispatch.mock.calls[0][0].tempo).toBe(mockEvent.target.value);
    });
  });

  describe('setBeatNumber', () => {
    const mockDispatch = jest.fn(() => {});
    const { setBeatNumber } = mapDispatchToProps(mockDispatch);
    
    it('calls dispatch method once', () => {
      setBeatNumber(mockEvent);
      expect(mockDispatch.mock.calls.length).toBe(1);
    });

    it('dispatches action with correct value', () => {
      mockDispatch.mockReset();
      setBeatNumber(5);
      expect(mockDispatch.mock.calls[0][0].value).toBe(5);
    });
  });
});