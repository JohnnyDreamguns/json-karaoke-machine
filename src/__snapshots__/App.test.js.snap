// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg navbar-light fixed-top"
    id="mainNav"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand js-scroll-trigger"
        href="/"
      >
        JSON Karaoke Machine
      </a>
    </div>
  </nav>
  <header
    className="masthead"
  >
    <div
      className="container h-100"
    >
      <div
        className="row h-100"
      >
        <div
          className="col-lg-7 my-auto"
        >
          <div
            className="header-content mx-auto"
          >
            <h1
              className="mb-5"
            >
              Choose a song and press play to hear the sweet sounds of the Web Audio API
            </h1>
            <div
              className="header-controls"
            >
              <button
                aria-checked="false"
                className="btn btn-outline btn-xl"
                data-playing="false"
                onClick={[Function]}
                role="switch"
              >
                Play!
              </button>
              <div
                className="tempo-control"
              >
                <label
                  forhtml="tempo"
                >
                  bpm
                </label>
                <input
                  id="tempo"
                  max="250"
                  min="70"
                  step="1"
                  type="range"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <section
    className="bg-primary text-center instrument"
  >
    <h2>
      Drums
    </h2>
    <div
      className="json-display"
    >
      <div
        className="json"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "[<br><br>]",
            }
          }
        />
      </div>
      <div>
        <img
          alt=""
          className="img-fluid"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </section>
  <section
    className="features instrument"
  >
    <h2>
      Bass
    </h2>
    <div
      className="json-display"
    >
      <div
        className="json"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "[<br><br><br>]",
            }
          }
        />
      </div>
      <div>
        <img
          alt=""
          className="img-fluid"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </section>
  <section
    className="bg-primary text-center instrument"
  >
    <h2>
      PolySynth
    </h2>
    <div
      className="json-display"
    >
      <div
        className="json"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "[<br><br><br><br>]",
            }
          }
        />
      </div>
      <div>
        <img
          alt=""
          className="img-fluid"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </section>
  <footer>
    <div
      className="container"
    >
      <p>
        Â© JSON Karaoke Machine 2019. All Rights Reserved.
      </p>
    </div>
  </footer>
</div>
`;
